unblob/processing.py

```diff
@@ -287,6 +289,137 @@ def _process_task(self, result: TaskResult, task: Task):
+    def process(self):
+        if self.dir_task.multi_file:
+            # The file has been processed as part of a previous MultiFile
+            return
+
+        logger.debug("Processing directory", path=self.dir_task.path)
+
+        try:
+            processed_paths, extract_dirs = self._process_directory()
+        except DirectoryProcessingError:
+            return
+
+        self._iterate_directory(extract_dirs, processed_paths)
+
+        self._iterate_processed_files(processed_paths)
+
+    def _process_directory(self) -> Tuple[Set[Path], Set[Path]]:
+        processed_paths: Set[Path] = set()
+        extract_dirs: Set[Path] = set()
+        for dir_handler_class in self.config.dir_handlers:
+            dir_handler = dir_handler_class()
+
+            for path in dir_handler.PATTERN.get_files(self.dir_task.path):
+                multi_file = dir_handler.calculate_multifile(path)
+
+                if multi_file is None:
+                    continue
+
+                multi_file.handler = dir_handler
+
+                self._check_conflicting_files(multi_file, processed_paths)
+
+                extract_dir = self._extract_multi_file(multi_file)
+
+                # Process files in extracted directory
+                if extract_dir.exists():
+                    self.result.add_subtask(
+                        Task(
+                            extractable_id=multi_file.id,
+                            path=extract_dir,
+                            depth=self.dir_task.depth + 1,
+                        )
+                    )
+                    extract_dirs.add(extract_dir)
+
+                processed_paths.update(multi_file.paths)
+        return processed_paths, extract_dirs
+
+    def _check_conflicting_files(
+        self, multi_file: MultiFile, processed_paths: Set[Path]
+    ):
+        conflicting_paths = processed_paths.intersection(set(multi_file.paths))
+        if conflicting_paths:
+            logger.error(
+                "Conflicting match on files",
+                handler=multi_file.handler.NAME,
+                directory=self.dir_task.path,
+                paths=conflicting_paths,
+            )
+            self.result.add_report(
+                MultiFileCollisionReport(
+                    paths=conflicting_paths, handler=multi_file.handler.NAME
+                )
+            )
+            raise DirectoryProcessingError
+
+    def _extract_multi_file(self, multi_file: MultiFile) -> Path:
+        extract_dir = self.config.get_extract_dir_for(
+            self.dir_task.path / multi_file.name
+        )
+        if extract_dir.exists():
+            logger.error(
+                "Skipped: extraction directory exists",
+                extract_dir=extract_dir,
+                multifile=multi_file,
+            )
+            self.result.add_report(
+                multi_file.as_report([ExtractDirectoryExistsReport(path=extract_dir)])
+            )
+            raise DirectoryProcessingError
```

Comment:
Again: just raise the error with the details, no logging, no storing in results.